### ISSUE TITLE
Graceful and consistent handling of non-text clipboard instead of error message

### ISSUE THREAD

**Issue by woerndl** [federico-terzi/espanso#2075]
### Description

I often have non-text content lingering in my clipboard. This could be image data or elements from Figma/Sketch etc. I think the behavior for an empty clipboard already changed since 2.1.8, when I first raised the [issue on Reddit](https://www.reddit.com/r/espanso/comments/16kv1rr/comment/k2ftlt8/).

But when I have image data in the clipboard or other non-text content expansions still result in:

```
[Espanso]: An error occurred during rendering, please examine the logs for more information.
```

My expected behavior for the example trigger `<h1>{{clipboard}}</h1>` would be:

```
# Example with non-text clipboard to fill in later
<h1></h1>

# Example with text clipboard content
<h1>Text clipboard content</h1>

# Instead of
[Espanso]: An error occurred during rendering, please examine the logs for more information.
```

When there is a file in the clipboard on the other hand you get:

```
<h1>filename.txt</h1>
```

Which can be handy for paths or similar, but is somewhat inconsistent behavior.

### Motivation

As a user coming from TextExpander and aText, this is rather surprising and inconvenient to me, as I often use clipboard snippets without paying attention to the clipboard content. Additionally, you have to delete the error message manually, since you can’t use backspace to remove it. A small mistake creates quite a hassle, disrupting the workflow.

This is probably why both aText and TextExpander handle non-text clipboard content as an empty string, as I described.

I rely heavily on clipboard snippets, and this behavior has kept me from being able to productively use Espanso, which is unfortunate because the project has so much potential, and I really like the file-based configs.

### Alternatives

It might even make sense to turn the handling of non-text clipboard content into a setting.

### Have you tried building it? Or would you like it?

- [ ] I'd be willing to contribute this feature to Espanso myself.

**Comment by smeech** [comment]
Just for clarity, I presume you're using code like:
```yml
  - trigger: :test
    replace: <h1>{{clipboard}}</h1>
    vars:
      - name: clipboard
        type: clipboard
```
I agree, Espanso error-handling is rather clunky - referring users to the log, e.g.:
```
13:28:01 [worker(488753)] [WARN] extension 'clipboard' on var: 'clipboard' reported an error: clipboard provider returned error
13:28:01 [worker(488753)] [ERROR] error during rendering: rendering error

Caused by:
    clipboard provider returned error
```
in this image case, rather than showing the error itself.

I guess user-preference will vary between those who would like to see that something unexpected has happened, and those who would just like a semi-usable outcome like `<h1></h1>`. At least the clipboard file-name handling is graceful!

**Comment by woerndl** [comment]
Yes, your clarifying example is what I meant.

> Espanso error-handling is rather clunky

I agree, but in this case, it goes beyond error handling—it actively breaks the workflow by expanding the error message itself, which leaves users with more work to fix. Not only does it disrupt the process, but it also forces users to manually delete the error message. This makes the onboarding experience with the clipboard extension unnecessarily bad, as most regular users will run into this issue fairly quickly.

> I guess user-preference will vary

True, but I’d argue the default should be expanding as if the clipboard were empty. This is the expected behavior in other tools like TextExpander and aText. Expanding to an empty string would be much more seamless for most users, especially for those who rely on clipboard snippets without checking the content every time.

For the minority with edge cases (e.g., where an empty clipboard resembles a previous error), there could be a setting to enable strict mode for those who prefer an error message.

**Comment by smeech** [comment]
I agree - a sound and an escapable error pop-up (perhaps containing the actual error) would make more sense.
